app.ts

```diff
@@ -0,0 +1,412 @@
+class DataHandler {
+	/** It tracks if the showRecords function is running and would the value would be false if it's not. */
+	private isFunctionRunning: boolean;
+	/** This is an instance of the ApiManager class and passing a string when being used. */
+	private apiManager: ApiManager;
+	/** The following track what page the user is on when navigating through the web app. */
+	private currentPage: number;
+	/** This tracks the first page of the pagination that is being displayed. */
+	private paginationStart: number;
+	/** This will track the last page that is being displayed. */
+	private paginationEnd: number;
+	/** Tracks the current starting ID in the table. */
+	private currentFromID: number;
+	/** Track the current last ID in the table. */
+	private currentToID: number;
+	/** This is the calculation for the difference between the original ID  based on the currentPage
+	 * and the currentID which is based on the resizing of the table. */
+	private difference: number;
+	/** A timer that handles window resizing events and store the return type of setTimeout. */
+	private resizeTimer: ReturnType<typeof setTimeout> | null;
+
+	constructor() {
+		this.isFunctionRunning = false;
+		this.apiManager = new ApiManager("http://localhost:2050/");
+		this.currentPage = 1;
+		this.paginationStart = 1;
+		this.paginationEnd = 10;
+		this.currentFromID = 0;
+		this.currentToID = 20;
+		this.difference = 0;
+		this.resizeTimer = null;
+	}
+
+	/** Returning columns and rendering it on the table in the dom. */
+	showColumns(): Promise<void> {
+		$(".head-row").empty();
+		return this.apiManager.getColumns()
+			.then(columns => {
+				for (const column of columns) {
+					$("#records-table-heading").append(`<th>${column}</th>`);
+				}
+			})
+			.catch(error => {
+				throw new Error(`Failed to retrieve columns: ${error}`)
+			});
+	}
+
+	/** Fetching the records and rendering it on the DOM in the table*/
+	showRecords(fromID: number, toID: number): Promise<void> {
+		if (this.isFunctionRunning) {
+			return Promise.resolve(undefined);
+		}
+		this.isFunctionRunning = true;
+		let inputNumber: string;
+		let stringCount: string;
+		return this.apiManager.getRecordCount()
+			.then(count => {
+				inputNumber = this.input();
+				const maxRecords = this.recordsPerPage();
+				$("#records-table-body").empty();
+				this.loader();
+				this.currentToID = toID
+				this.currentFromID = fromID;
+				if (toID >= (count - 1)) {
+					this.currentToID = (count - 1);
+					this.currentFromID = this.currentToID - (maxRecords - 1);
+				}
+				else if (this.currentPage === 1) {
```

Comment:
```suggestion
				} else if (this.currentPage === 1) {
```
On the same line please